cmd/tailscale/cli/file.go

```diff
@@ -449,9 +442,99 @@ func runFileGet(ctx context.Context, args []string) error {
 		}
 		deleted++
 	}
-	if getArgs.verbose {
+	if deleted == 0 && len(wfs) > 0 {
+		// persistently stuck files are basically an error
+		errs = append(errs, fmt.Errorf("moved %d/%d files", deleted, len(wfs)))
+	} else if getArgs.verbose {
 		printf("moved %d/%d files\n", deleted, len(wfs))
 	}
+	return errs
+}
+
+type errorMemory struct {
+	firstSeen    time.Time
+	lastReported time.Time
+}
+
+type groundhogsDayLogger struct {
+	stutterTime time.Duration // log persistent errors at this interval if > 0.
+	log         map[string]*errorMemory
+}
+
+// For every new batch of data, are the errors we see any different from last time?
+// If so, say how they changed.  If not, they're not new errors, so don't bother logging them.
+func (gdl *groundhogsDayLogger) NextBatch(errs []error) {
+	t := time.Now()
+	newlog := make(map[string]*errorMemory)
+	for _, err := range errs {
+		s := err.Error()
+		oldentry, exist := gdl.log[s]
+		if exist {
+			newlog[s] = oldentry
+		} else {
+			newlog[s] = &errorMemory{t, t}
+			outln(s)
+		}
+	}
+	for k, _ := range gdl.log {
+		if _, still_exists := newlog[k]; !still_exists {
+			outln("Error cleared: ", k)
+		}
+	}
+	gdl.log = newlog
+}
+
+// Make sure persistent errors gets flushed again when their stutter time has passed.
+func (gdl *groundhogsDayLogger) PeriodicFlush() {
+	if gdl.stutterTime <= 0 {
+		return
+	}
+	t := time.Now()
+	for k, v := range gdl.log {
+		if t.Sub(v.lastReported) > gdl.stutterTime {
+			outln("Persistent error for", t.Sub(v.firstSeen), ":", k)
+			v.lastReported = t
+		}
+	}
+}
```

Comment:
Let's delete all this new log limiting code for now and just get this merged in the basic form that logs without limits.

Reply:
done.